docs/LARGE_TESTS.md

```diff
@@ -0,0 +1,165 @@
+      - "8083"
+      - “8086"
+```
+### Travis CI:
+
+To enable large tests on Travis CI, please enable sudo, docker, and add the appropriate test matrix settings in `.sync.yml`:
+```
+.travis.yml:
+  sudo: true # large tests require travis.ci VMs instead of containers (enabled via sudo: true)
+  services:  # this ensures docker/docker-compose is installed on the travis agent
+    - docker
+  env:
+    global:   # If you change the matrix, please preserve environment globals:
+      - ORG_PATH=/home/travis/gopath/src/github.com/intelsdi-x
+      - SNAP_PLUGIN_SOURCE=/home/travis/gopath/src/github.com/${TRAVIS_REPO_SLUG}
+    matrix:
+      - TEST_TYPE: small             # preserve existing small tests
+      - TEST_TYPE: medium            # preserve existing medium tests (make sure they exist)
+      # if SNAP_VERSION:latest and OS:alpine is sufficient simply add TEST_TYPE: large
+      - TEST_TYPE: large
+      # if multiple SNAP_VERSION, OS needs to be tested, provide an array of versions:
+      - SNAP_VERSION=latest OS=xenial TEST_TYPE=large
+      - SNAP_VERSION=latest_build OS=centos7 TEST_TYPE=large
+  matrix:
+    # travis doesn't have an easy way to exclude large tests with a regex, so
+    # please list every large test to exclude it from running on go 1.6.x
+    exclude:
+      - go: 1.6.x
+        env: TEST_TYPE=large
+      - go: 1.6.x
+        env: SNAP_VERSION=latest OS=xenial TEST_TYPE=large
+      - go: 1.6.x
+        env: SNAP_VERSION=latest_build OS=centos7 TEST_TYPE=large
+```
+NOTE: If you did not set `sudo: true` and enable docker services, in travis.ci large test will fail with the following error:
+```
+2017-02-06 23:00:35 UTC [    error] docker needs to be installed
+```
+
+### Serverspec
+
+The large tests are written using serverspec as the system test framework. An example installing and testing `ping`:
+```
+set :docker_compose_container, :snap                       # required if you use the os["family"] detection functionality
+
+context "network is functional" do
+  if os["family"] == "ubuntu"
+    describe package("iputils-ping") do
+      it { should be_installed }
+    end
+  elsif os["family"] == "redhat"
+    describe package("iputils") do
+      it { should be_installed }
+    end
+  end
+
+  describe command('ping -c1 8.8.8.8') do
+    its(:exit_status) { should eq 0 }
+    its(:stdout) { should contain(/1 packets received/) }
+  end
+end
+```
+
+If you have more than one container specified in docker compose, tests can be executed in each container separately:
+```
+describe docker_compose('./docker_compose.yml') do
+  its_container(:snap) do
+    # these tests would only run in the snap container
+  end
+
+  its_container(:influxdb) do
+    # these tests would only run in the influxdb container
+  end
+end
+```
+
+## Running Tests
+In addition to `make test-large` which is described in [BUILD_AND_TEST.md](BUILD_AND_TEST.md), you have the additional following options when using the large test framework:
```

Comment:
Yes, I think people should explore that link at this point in the sentence. Because in BUILD_AND_TEST it describes the basics of how to run `make test-xx` where `xx` is `small`, `medium`, or `large`. Below this line describes more advanced variables you can supply in the command that are only available for large tests.